microk8s-resources/actions/enable.cilium.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+
+if ! which helm 2>&1 >/dev/null; then
+	echo "Cilium installation requires helm. Please install via 'snap install helm'." >&2
+	exit 1
+fi
+
+ARCH=$(arch)
+if ! [ "${ARCH}" = "amd64" ]; then
+  echo "Cilium is not available for ${ARCH}" >&2
+  exit 1
+fi
+
+echo "Enabling Cilium"
+
+if [ ! -f "${SNAP_DATA}/bin/cilium" ]
+then
+  CILIUM_VERSION="master"
+  CILIUM_ERSION=$(echo $CILIUM_VERSION | sed 's/v//g')
+  CILIUM_DIR="cilium-$CILIUM_ERSION"
+  SOURCE_URI="https://github.com/cilium/cilium/archive"
+  CILIUM_CNI_CONF="plugins/cilium-cni/05-cilium-cni.conf"
+  CILIUM_LABELS="k8s-app=cilium"
+  NAMESPACE=kube-system
+
+  echo "Fetching cilium version $CILIUM_VERSION."
+  sudo mkdir -p "${SNAP_DATA}/tmp/cilium"
+  (cd "${SNAP_DATA}/tmp/cilium"
+  sudo "${SNAP}/usr/bin/curl" -L $SOURCE_URI/$CILIUM_VERSION.tar.gz -o "$SNAP_DATA/tmp/cilium/cilium.tar.gz"
+  sudo gzip -f -d "$SNAP_DATA/tmp/cilium/cilium.tar.gz"
+  sudo tar -xf "$SNAP_DATA/tmp/cilium/cilium.tar" "$CILIUM_DIR/install" "$CILIUM_DIR/$CILIUM_CNI_CONF")
+
+  sudo mv "$SNAP_DATA/args/cni-network/cni.conf" "$SNAP_DATA/args/cni-network/10-kubenet.conf" 2>/dev/null || true
+  sudo cp "$SNAP_DATA/tmp/cilium/$CILIUM_DIR/$CILIUM_CNI_CONF" "$SNAP_DATA/args/cni-network/05-cilium-cni.conf"
+
+  (cd "${SNAP_DATA}/tmp/cilium/$CILIUM_DIR/install/kubernetes"
+  helm template cilium \
+      --namespace $NAMESPACE \
+      --set global.cni.confPath="$SNAP_DATA/args/cni-network" \
+      --set global.cni.customConf=true \
+      --set global.containerRuntime.integration="containerd" \
+      --set global.containerRuntime.socketPath="$SNAP_COMMON/run/containerd.sock" \
+      | sudo tee cilium.yaml >/dev/null)
+
+  sudo mkdir -p "$SNAP_DATA/actions/cilium/"
+  sudo cp "$SNAP_DATA/tmp/cilium/$CILIUM_DIR/install/kubernetes/cilium.yaml" "$SNAP_DATA/actions/cilium.yaml"
+
+  "$SNAP/kubectl" "--kubeconfig=$SNAP/client.config" apply -f "$SNAP_DATA/actions/cilium.yaml"
```

Comment:
Please take the kubeconfig file found under credentials (`--kubeconfig=$SNAP_DATA/credentials/client.config`), the one you are using is kept there for backwards compatibility reasons and will soon be removed.